diff -r f01b306ebe0e -r f5afcce5f1dc pager.c
--- a/pager.c	Thu Dec 30 10:09:37 2010 -0800
+++ b/pager.c	Sat Feb 12 11:29:00 2011 +0900
@@ -2766,6 +2766,7 @@ search_next:
   if (IsHeader (extra))
   {
     Context->msgnotreadyet = -1;
+    /*
     if (rc == -1)
       OldHdr = NULL;
     else
@@ -2773,6 +2774,7 @@ search_next:
       TopLine = topline;
       OldHdr = extra->hdr;
     }
+    */
   }
     
   cleanup_quote (&QuoteList);