diff --git a/getmailcore/_retrieverbases.py b/getmailcore/_retrieverbases.py
index 46dc856..24fac17 100755
--- a/getmailcore/_retrieverbases.py
+++ b/getmailcore/_retrieverbases.py
@@ -318,6 +318,7 @@ class RetrieverSkeleton(ConfigurableBase):
     def __init__(self, **args):
         self.msgnum_by_msgid = {}
         self.msgid_by_msgnum = {}
+        self._msgid_by_msgnum_sorted = None
         self.msgsizes = {}
         self.headercache = {}
         self.oldmail = {}
@@ -345,7 +346,11 @@ class RetrieverSkeleton(ConfigurableBase):
         self.log.trace('i == %d' % i)
         if not self.__initialized:
             raise getmailOperationError('not initialized')
-        return sorted(self.msgid_by_msgnum.items())[i][1]
+        if self._msgid_by_msgnum_sorted is None:
+            msgid_by_msgnum_sorted = sorted(self.msgid_by_msgnum.items())
+        else:
+            msgid_by_msgnum_sorted = self._msgid_by_msgnum_sorted
+        return msgid_by_msgnum_sorted[i][1]
 
     def _read_oldmailfile(self):
         '''Read contents of oldmail file.'''
@@ -473,6 +478,7 @@ class POP3RetrieverBase(RetrieverSkeleton):
         return self.msgnum_by_msgid[msgid]
 
     def _getmsglist(self):
+        self._msgid_by_msgnum_sorted = None
         self.log.trace()
         try:
             response, msglist, octets = self.conn.uidl()
@@ -521,6 +527,7 @@ class POP3RetrieverBase(RetrieverSkeleton):
                 'command, use BrokenUIDLPOP3Retriever instead'
                  % o
             )
+        self._msgid_by_msgnum_sorted = sorted(self.msgid_by_msgnum.items())
         self.gotmsglist = True
 
     def _delmsgbyid(self, msgid):
