commit 94e78f0482c1abb1314966ac9e51f135ce986072
Author: NAKAMURA Yoshitaka <arumakanoy@gmail.com>
Date:   Fri Feb 18 16:43:52 2011 +0900

    add more keybind settings

diff --git a/data/defaults.cfg b/data/defaults.cfg
index c3da557..e80bf37 100644
--- a/data/defaults.cfg
+++ b/data/defaults.cfg
@@ -26,6 +26,13 @@ show_help = F1
 show_preferences = <Control>semicolon
 switch_to_source = <Control>s
 toggle_text_mode_quick = <Control>period
+current_go_down = <Control>n
+current_go_up = <Control>p
+switch_current = <Control>i
+switch_current_reverse = <Shift><Control>i
+browse_down = <Control>f
+browse_back = <Control>b
+cancel = <Control>g
 
 
 # Catalog: The sources of the plugin are included
diff --git a/kupfer/ui/browser.py b/kupfer/ui/browser.py
index 7333610..bac5996 100644
--- a/kupfer/ui/browser.py
+++ b/kupfer/ui/browser.py
@@ -1395,6 +1395,39 @@ class Interface (gobject.GObject):
 		self.data_controller.search(pane, key=text, context=text,
 				text_mode=self.get_in_text_mode())
 
+	def current_go_up(self):
+		self.current.go_up()
+
+	def current_go_page_up(self):
+		self.current.go_page_up()
+
+	def current_go_down(self):
+		if (not self.current.get_current() and
+				self.current.get_match_state() is State.Wait):
+			self._populate_search()
+		self.current.go_down()
+
+	def current_go_page_down(self):
+		if (not self.current.get_current() and
+				self.current.get_match_state() is State.Wait):
+			self._populate_search()
+		self.current.go_page_down()
+
+	def current_go_first(self):
+		self.current.go_first()
+
+	def switch_current_reverse(self):
+		self.switch_current(reverse=True)
+
+	def browse_down(self):
+		self._browse_down()
+
+	def browse_back(self):
+		self._back_key_press()
+
+	def cancel(self):
+		self._escape_key_press()
+
 gobject.type_register(Interface)
 gobject.signal_new("cancelled", Interface, gobject.SIGNAL_RUN_LAST,
 		gobject.TYPE_BOOLEAN, ())
diff --git a/kupfer/ui/preferences.py b/kupfer/ui/preferences.py
index ce57104..76d82bc 100644
--- a/kupfer/ui/preferences.py
+++ b/kupfer/ui/preferences.py
@@ -79,6 +79,13 @@ class PreferencesWindowController (pretty.OutputMixin):
 		'show_preferences': _('Show Preferences'),
 		'switch_to_source': _('Switch to First Pane'),
 		"toggle_text_mode_quick": _('Toggle Text Mode'),
+		'current_go_down': 'Go Down',
+		'current_go_up': 'Go Up',
+		'switch_current': 'Switch Current',
+		'switch_current_reverse': 'Switch Current Reverse',
+		'browse_down': 'Browse Down',
+		'browse_back': 'Browse Back',
+		'cancel': 'Cancel',
 	}
 
 	def __init__(self):
